### PR TITLE
Fixed rank dtypes issues

### LINKED ISSUE #201: rank throws RayGetError
### Describe the problem
`rank` throws a `RayGetError`

### Source code / logs
```
import modin.pandas as pd

data = {
    "col3": 1.0,
    "col4": np.datetime64("2011-06-15T00:00"),
    "col5": np.array([3] * 4, dtype="int32"),
    "col1": "foo",
    "col2": True,
}

modin_df = pd.DataFrame(data)
modin_df.rank(numeric_only=False, axis=1)
```

### PR DESCRIPTION
**Pull request by williamma12**
## What do these changes do?

Does dtype checking before passing to `data_manager`

## Related issue number

Resolves #201 

- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] passes `black --check modin/`

### CODE REVIEW

File: modin/data_management/data_manager.py [hunk 1]
```diff
@@ -1792,15 +1792,21 @@ def rank(self, **kwargs):
...
         func = self._prepare_method(pandas.DataFrame.rank, **kwargs)
-        new_data = self.map_across_full_axis(axis, func)
-        # Since we assume no knowledge of internal state, we get the columns
-        # from the internal partitions.
         if numeric_only:
-            new_columns = self.compute_index(1, new_data, True)
+            result, data_manager = self.numeric_function_clean_dataframe(axis)
+            if result is not None:
+                return self.from_pandas(
+                    pandas.DataFrame(index=data_manager.index), type(self.data)
```
Comment (devin-petersohn) [inline modin/data_management/data_manager.py]: how is `data_manager.index` chosen here?

Comment (williamma12) [reply modin/data_management/data_manager.py]: `data_manager.index` is the same as `self.index`. I changed it to `self.index` for readability

**Comment by williamma12** [comment]
Blocked on #241

**Comment by williamma12** [comment]
No longer doing dtype checking

**Pull request closed without merging**